File: src/main/java/hello/controller/GreetingController.java
package hello.controller;

import hello.bean.Greeting;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Created by dongxie on 2017/5/7.
 */
@RestController
public class GreetingController {

    private static final String template = "Hello, %s!";
    private static Logger log = LoggerFactory.getLogger(GreetingController.class);
    private final AtomicLong counter = new AtomicLong();

    @RequestMapping("/greeting")
    public Greeting greeting(@RequestParam(value = "name", defaultValue = "World") String name) {
        log.info("this is run");
        return new Greeting(counter.incrementAndGet(),
                String.format(template, name));
    }


}

File: src/main/java/hello/controller/GreetingController.kt
package hello.controller

import hello.bean.Greeting
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.util.concurrent.atomic.AtomicLong

/**
 * Created by dongxie on 2017/5/7.
 */
@RestController
class GreetingController {
    private val counter = AtomicLong()

    @RequestMapping("/greeting")
    fun greeting(@RequestParam(value = "name", defaultValue = "World") name: String): Greeting {
        log.info("this is run")
        return Greeting(counter.incrementAndGet(),
                String.format(template, name))
    }

    companion object {

        private val template = "Hello, %s!"
        private val log = LoggerFactory.getLogger(GreetingController::class.java)
    }


}
